schemas/2015-08-01/Microsoft.Cache.json

```diff
@@ -0,0 +1,160 @@
+                                            "type": "integer"
+                                        },
+                                        {
+                                            "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                                        }
+                                    ],
+                                    "description": "Microsoft.Cache/Redis: along with family, determines capacity of the cache",
+                                    "default": 1
+                                }
+                            },
+                            "required": [
+                                "name",
+                                "family",
+                                "capacity"
+                            ]
+                        },
+                        "shardCount": {
+                            "oneOf": [
+                                {
+                                    "type": "integer"
+                                },
+                                {
+                                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                                }
+                            ],
+                            "description": "Microsoft.Cache/Redis: (only if using clustering) number of shards in the redis cluster"
+                        },
+                        "enableNonSslPort": {
+                            "oneOf": [
+                                {
+                                    "type": "boolean"
+                                },
+                                {
+                                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                                }
+                            ],
+                            "description": "Microsoft.Cache/Redis if set to true, enables less secure direct access to redis on port 6379 (WITHOUT SSL tunneling)"
+                        },
+                        "redisConfiguration": {
+                            "type": "object"
+                        },
+                        "virtualNetwork": {
+                            "oneOf": [
+                                {
+                                    "$ref": "#/definitions/virtualNetworkId"
+                                },
+                                {
+                                    "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                                }
+                            ],
+                            "description": "Microsoft.Cache/Redis (optional) the ARM resource ID of a Classic Virtual Network for the redis cache to join"
+                        },
+                        "subnet": {
```

Comment:
I believe subnets are a child-resource of virtualNetworks, right? In that case, could you get rid of "virtualNetwork" and just ask the deployment template author to provide a subnet resource uri, which would contain the virtual network?